plugins/woocommerce/src/Blocks/Patterns/PTKClient.php

```diff
@@ -14,6 +14,71 @@ class PTKClient {
 	 */
 	const PATTERNS_TOOLKIT_URL = 'https://public-api.wordpress.com/rest/v1/ptk/patterns/';
 
+	/**
+	 * The schema for the patterns toolkit.
+	 *
+	 * @var array
+	 */
+	private $schema;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->schema = [
+			'type'     => 'array',
+			'items'    => [
+				'type'       => 'object',
+				'properties' => [
+					'ID'           => [
+						'type' => 'integer',
+					],
+					'site_id'      => [
+						'type' => 'integer',
+					],
+					'title'        => [
+						'type' => 'string',
+					],
+					'name'         => [
+						'type' => 'string',
+					],
+					'html'         => [
+						'type' => 'string',
+					],
+					'categories'   => [
+						'type'                 => 'object',
+						'additionalProperties' => [
+							'type'       => 'object',
+							'properties' => [
+								'slug'  => [
+									'type' => 'string',
+								],
+								'title' => [
+									'type' => 'string',
+								],
+							],
+							'required'   => [ 'slug', 'title' ],
+						],
+					],
+					'dependencies' => [
+						'type'  => 'array',
+						'items' => [
+							'type' => 'string',
+						],
+					],
+				],
+			],
+			'required' => [
```

Comment:
WP schema validator handles required prop differnelty, it must be defined at the prop level not on a separate one

> Version 3 Syntax
While WordPress mostly follows JSON Schema Version 4, one way that it doesn’t is with the syntax for defining required properties. The predominant way is using JSON Schema Version 3 syntax by adding the required keyword to each property’s definition.
https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/

Reply:
Oh nice catch! I fixed it with fed56eca1f77c5b5d394a950901ff3aedf2aad81